contracts/TradeTypesLib2.sol

```diff
@@ -0,0 +1,134 @@
+/*
+ * Hubii Nahmii
+ *
+ * Compliant with the Hubii Nahmii specification v0.12.
+ *
+ * Copyright (C) 2017-2018 Hubii AS
+ */
+
+pragma solidity >=0.4.25 <0.6.0;
+
+import {MonetaryTypesLib} from "./MonetaryTypesLib.sol";
+import {NahmiiTypesLib} from "./NahmiiTypesLib.sol";
+
+/**
+ * @title     TradeTypesLib
+ * @dev       Data types centered around trade
+ */
+library TradeTypesLib2 {
+    //
+    // Enums
+    // -----------------------------------------------------------------------------------------------------------------
+    enum CurrencyRole {Intended, Conjugate}
+    enum LiquidityRole {Maker, Taker}
+    enum Intention {Buy, Sell}
+    enum TradePartyRole {Buyer, Seller}
+
+    //
+    // Structures
+    // -----------------------------------------------------------------------------------------------------------------
+    struct Operator {
+        uint256 id;
+        string data;
+    }
+
+    struct OrderPlacement {
+        Intention intention;
+
+        int256 amount;
+        NahmiiTypesLib.IntendedConjugateCurrency currencies;
+        int256 rate;
```

Comment:
How will the rate work? We need to be able to express rates both as 1000:1 and 1:1000 for instance

Reply:
One option here is to define rate in terms of an entirety of 10^18 rather than of 1 or 100.

Reply:
@jijordre, will that approach also work when the two currencies have different number of decimals?

Reply:
It will indeed. The number of decimals on currencies is independent on the basis for entirety of exchange rate.